gateway/ln-gateway/src/lib.rs

```diff
@@ -670,10 +670,22 @@ impl Gateway {
             address,
             federation_id,
         } = payload;
-
         let client = self.select_client(federation_id).await?;
-        // TODO: This should probably be passed in as a parameter
-        let fees = client.get_withdraw_fee(address.clone(), amount).await?;
+
+        // TODO: Fees should probably be passed in as a parameter
+        let (amount, fees) = match amount {
+            // If the amount is "all", then we need to subtract the fees from
+            // the amount we are withdrawing
+            BitcoinAmountOrAll::All => {
+                let balance = bitcoin::Amount::from_sat(client.get_balance().await.msats * 1000);
```

Comment:
I don't think there's any reason to round it down here. It should all work on msats, no?

Reply:
Yeah, `fedimint_core::Amount` has a higher precision, and if user has `1000.005sats`, fee is `1000sats`, they should be able to widraw `0.005msat` just fine, no matter how uneconomical it is. :D

Reply:
For an onchain pegout transaction you can't get down to msats, it has to be nearest sat?

Reply:
:brain::hammer: 

Right. :D

Thought I'd still keep it in highest precision available&needed, and round down at the very last place where it is necessary as a best practice. Otherwise the code e.g. ever changes it might lead to a mistake.

But not a blocker.

Reply:
amount isn't used anywhere past withdraw, should we rewrite withdraw to take a FedimintAmount instead of a BitcoinAmount then?

Reply:
Nah. Whateva.